tranco/tranco.py

```diff
@@ -11,16 +11,13 @@ def __init__(self, date, list_id, lst):
         self.date = date
         self.list_id = list_id
         self.list_page = "https://tranco-list.eu/list/{}/1000000".format(list_id)
-        self.list = lst
+        self.list = {domain: index + 1 for index, domain in enumerate(lst)}
 
     def top(self, num=1000000):
-        return self.list[:num]
+        return [domain for domain in sorted(self.list, key=self.list.get)][:num]
```

Comment:
```suggestion
        return sorted(self.list, key=self.list.get)[:num]
```

`sorted` already returns a new list of sorted domains, so we can reduce execution time further by removing this list comprehension.